Data/Text/ICU/CaseMap.hsc

```diff
@@ -56,7 +57,7 @@ reduceCaseMapOptions = foldl' (.|.) (#const U_FOLD_CASE_DEFAULT) .
                        map fromCaseMapOption
 
 caseMap :: LocaleName -> [CaseOption] -> CaseMap
-caseMap name opts = unsafePerformIO $ do
+caseMap name opts = unsafePerformIO $ mask_ $ do
```

Comment:
why the addition of `mask_` here?

Reply:
It's to prevent memory leaks. The C data structure is allocated in the `ucasemap_open` call, and for a short time (before `newForeignPtr` is called), it lives without a finalizer. If any asynchronous exception (such as a timeout) is raised during this time, the C data won't be deallocated. Rare case but it still needs to be handled.

I've added `mask_` in all places where `newForeignPtr` is called.

Reply:
Great idea. Done in 9abac52dbe9fcbf43c72a0b48d64150cffb9d671